reltime/reltime.py

```diff
@@ -463,39 +517,204 @@ def ground_text(self, text, base_date):
+        self.regex = REG12
 
-# ---- PUBLIC METHODS ---- #
+    def ground_text(self, text, base_date):
+        """Ground text that matches regex 12."""
+        # split up tag and grab relevant parts
+        time, am_pm = split_am_pm(text)
+        # split up numerical portion
+        hour, minute = split_hour_minute(time)
+        # apply to date as appropriate
+        if ((am_pm == "pm") and (hour != 12)):
+            hour += 12
+        if hour < base_date.hour:
+            day = base_date.day + 1
+        else:
+            day = base_date.day
+        new_date = base_date.replace(day=day, hour=hour, minute=minute)
+        return new_date
 
 
-def tag(text):
-    """Tag time info in text.
+class TimeNoAmPm:
+    """Class for time text referring to time without am/pm."""
+
+    def __init__(self):
+        """Initialize class."""
+        self.regex = REG13
+
+    def ground_text(self, text, base_date):
+        """Ground text that matches regex 13."""
+        # need to find the subset of the match that contains the numbers.
+        # this could be in several locations depending on the match
+        split_time = re.findall(r'((\d{1,2})(:\d{1,2})?)', text)
+        time = split_time[0][0]  # first match, full match.
+        # split up numerical portion
+        hour, minute = split_hour_minute(time)
+        # as a default, assume hours less than 10 are PM.
+        if hour < 10:
+            hour += 12
+        if hour < base_date.hour:
+            day = base_date.day + 1
+        else:
+            day = base_date.day
+        new_date = base_date.replace(day=day, hour=hour, minute=minute)
+        return new_date
+
+
+class TimeRange:
+    """Class for time text referring to time ranges."""
+
+    def __init__(self):
+        """Initialize class."""
+        self.regex = REG14
+
+    def ground_text(self, text, base_date):
+        """Ground text that matches regex 14."""
+        split_time = re.split(r'-', text)
+        parsed_time = []
+        for time in split_time:
+            time_text, ampm = split_am_pm(time)
+            time_dict = {'time': time_text, "ampm": ampm}
+            parsed_time.append(time_dict)
+        if ((parsed_time[0]['ampm'] is not "") & (parsed_time[1]['ampm'] is not "")):
+            for time in parsed_time:
+                hour, minute = split_hour_minute(time['time'])
+                if ((time['ampm'] == "pm") and (hour != 12)):
+                    hour += 12
+                time['hour'] = hour
+                time['minute'] = minute
+        elif ((parsed_time[0]['ampm'] is not "") | (parsed_time[1]['ampm'] is not "")):
+            ampm = parsed_time[0]['ampm'] + parsed_time[1]['ampm']
+            for time in parsed_time:
+                hour, minute = split_hour_minute(time['time'])
+                if ((ampm == "pm") and (hour != 12)):
+                    hour += 12
+                time['hour'] = hour
+                time['minute'] = minute
+        else:
```

Comment:
nope, 'ampm' is a key in the `time_dict` object. I'm thinking about making a class for this and one other object, but they seem small enough, temporary enough, and neither are public facing that it might not make sense.